uiowa/uiowa
# AoS Feature Revisited

Relates to #3837
Resolves #4034
Resolves #4041

This PR creates a 90% clean split of AoS functionality as it relates to Admissions. The AoS Feature Split has been stripped down and rebuilt. Additionally, a generic search block has been added to uiowa_core to allow off-page filtering of the AoS view page (eventually a views block).

## Stories

- As an Admissions visitor and site owner, I don't see a change in my AoS functionality and can still search AoS from the homepage.
- As a developer the only changes I see to Admissions is where `field_degree_abbreviation` calls options from as well as some updated display configuration (moderation block, etc.). The module is enabled but the feature split is inactive.
- As a developer I can enable the AoS split without issue.
- As a webmaster of a site with the AoS split on, I can create AoS content and have it show up at /areas-of-study. Depending on how I fill in the source link, link direct fields, I can see that the content in the view and teasers are appropriately linked.

## To Test

- pull and `composer install && blt frontend && blt ds && blt ds --site=admissions.uiowa.edu`
- Note default sync doesn't have any AoS config imported
- Note admissions sync only field_degree_abbreviation, split config and AoS display settings are updated as it relates to AoS.
- The default site, once the AoS split is enabled and imported now has degree_types, locations taxonomies and their corresponding fields.
- There is a separate Search Block that webmasters can place that can be wired up to send queries to /areas-of-study using the `title` query parameter or /search with the `terms` query parameter. The endpoint field should only allow autocomplete internal paths (/node/6) and relative links (/search), not external URLs (e.g. https://google.com)


<details>
<summary>completed todo</summary>
## TODO
- [x] #4034
- [x] https://github.com/uiowa/uiowa/issues/4041
  - [x] Block description, more help text? 
  - [x] Change endpoint to link field
    - [x] style autocomplete (doesn't meet contrast ratio)
    - [x] validate for internal/relative paths only? 
  - [x] Fix SearchBlock Form to construct the redirect using the endpoint and query.
  - [x] Style Tweaks?
- [x] SEO
- [x] Prettify
- [x] Account for or remove sticky at top of lists.
- [x] Adjust layout alignment so button doesn't move based on main content
- [x] Confirm path is /areas-of-study/[node:title]

</details>
